### PR TITLE
Fix invalid link of the commit status when ref is tag

### LINKED ISSUE #29731: Gitea actions reference invalid link for Git tag
### Description

Gitea actions link to broken git tag
image

**Actual**
http://100.64.65.25:3000/xxxx/src/branch/v24.3.12

**Expected**
http://100.64.65.25:3000/xxxx/src/tag/v24.3.12

### Gitea Version

1.21.7

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

UI bug

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

UI bug

### Database

None

### PR DESCRIPTION
**Pull request by yp05327**
Fix #29731
Caused by #24634
Also remove fixme.

ps: we can not fix the existed runs, as wrong refs are all recorded in DB, and we can not know whether they are branch or tag:
image

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by yp05327** [comment]
How to test notifier? 🤔

**Comment by lunny** [comment]
> How to test notifier? 🤔

We can trigger the event manually and check the database.

**Comment by yp05327** [comment]
It seems that we have no action integrate tests. Maybe this is a good start.

**Review by lunny — APPROVED** [review]

**Comment by yp05327** [comment]
So lucky, I would say, by the test I found another issue: 
delete tag will remove the ref first , then trigger a notify, so in rotifer it will got this error
image

So wait a minute, and let me fix it also.

**Comment by yp05327** [comment]
delete tag will trigger 3 events:
- push event, refs/tags/deleted
push commit ID: 0000000000000000000000000000000000000000
- delete event, empty ref
no ref provided, so default branch will be used
- release event, deleted action, refs/tags/deleted

So for the delete event, the expected ref is default branch, not the deleted tag?

**Review by wolfogre — APPROVED** [review]

**Comment by yp05327** [comment]
Tests are simple now.
And you can see the error mentioned above in tests locally:
image

**Comment by yp05327** [comment]
Tested in GitHub, delete a branch will only trigger delete event:
https://github.com/yp05327/test2/actions/workflows/testdelete.yml
https://github.com/yp05327/test2/blob/main/.github/workflows/testdelete.ym

File: services/actions/notifier.go [hunk 1]
```diff
@@ -623,6 +629,10 @@ func (n *actionsNotifier) UpdateRelease(ctx context.Context, doer *user_model.Us
 }
 
 func (n *actionsNotifier) DeleteRelease(ctx context.Context, doer *user_model.User, rel *repo_model.Release) {
+	if rel.IsTag {
```
Comment (lunny) [inline services/actions/notifier.go:632]: Is this really right? @wolfogre

Comment (yp05327) [reply services/actions/notifier.go:632]: I think this function make sense:
https://github.com/go-gitea/gitea/blob/17d7ab5ad4ce3d0fbc1251572c22687c237a30b1/services/release/release.go#L305-L370

``` golang
if delTag {
  ......
  notify_service.PushCommits()
  notify_service.DeleteRef()
}
......
if !rel.IsDraft {
  notify_service.DeleteRelease(ctx, doer, rel)
}
```

Comment (wolfogre) [reply services/actions/notifier.go:632]: I think it's fine, it doesn't affect the actual usage scenario.

**Review by wolfogre — APPROVED** [review]

**Pull request merged**